src/cls/rbd/cls_rbd_types.h

```diff
@@ -310,6 +310,41 @@ struct SnapshotNamespaceOnDisk {
 };
 WRITE_CLASS_ENCODER(SnapshotNamespaceOnDisk);
 
+enum DeletedImageSource {
+  USER = 0,
+  RBD_MIRROR = 1
+};
+
+inline void encode(const DeletedImageSource &source, bufferlist& bl,
+		   uint64_t features=0)
+{
+  ::encode(static_cast<uint8_t>(source), bl);
+}
+
+inline void decode(DeletedImageSource &source, bufferlist::iterator& it)
+{
+  uint8_t int_source;
+  ::decode(int_source, it);
+  source = static_cast<DeletedImageSource>(int_source);
+}
+
+struct DeletedImageSpec {
```

Comment:
Nit: perhaps to keep this semi-consistent w/ the class methods and the public APIs -- s/Deleted/Trash/g